x-pack/plugins/cross_cluster_replication/public/app/components/follower_index_form.js

```diff
@@ -333,62 +307,48 @@ export const FollowerIndexForm = injectI18n(
       /**
        * Leader index
        */
-      const renderLeaderIndex = () => {
-        const hasError = !!fieldsErrors.leaderIndex;
-        const isInvalid = hasError && areErrorsVisible;
+      const renderLeaderIndex = () => (
```

Comment:
What's the intention behind creating a function here instead of assigning the component instance?

```jsx
const leaderIndex = (
  <FormEntryRow
    field="leaderIndex"
    value={followerIndex.leaderIndex}
    error={fieldsErrors.leaderIndex}
    schema={followerIndexFormSchema.leaderIndex}
    disabled={!isNew}
    areErrorsVisible={areErrorsVisible}
    onValueUpdate={this.onFieldsChange}
    onErrorUpdate={this.onFieldsErrorChange}
  />
);
```

Or (even better IMO), why not render it inline on line 446? I have the same questions for similar functions like `renderAdvancedSettings`.

Reply:
I think the intent is just to compartmentalize different pieces of the UI so that the main `render()` is easily readable. I don't mind the usage of this pattern here, but I also don't feel strongly since the pieces are not complex